runners/google-cloud-dataflow-java/examples/build.gradle

```diff
@@ -56,3 +60,41 @@ task preCommit(type: Test) {
   maxParallelForks 4
   systemProperty "beamTestPipelineOptions", JsonOutput.toJson(preCommitBeamTestPipelineOptions)
 }
+
+project.rootProject.ext.set("docker-repository-root", "${dockerImageRoot}")
```

Comment:
When this project is evaluated, it will override any properties users may have configured.

You'll need to restructure the `buildAndPushDockerContainer` task such that you can get the image/tag as a property of the `:beam-sdks-java-container:docker` run.

Also, why are we not using `dockerPush` part of the gradle docker plugin (https://github.com/palantir/gradle-docker)?